### PR TITLE
WIP ceph-common: test mon initial members

### PR DESCRIPTION
**Pull request by leseb**
Use a more common description for monitors and remove the sub monitor sections.

Closes: #803 
Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
test this please

**Comment by leseb** [comment]
test this please

**Comment by leseb** [comment]
test this please

**Comment by leseb** [comment]
test this please

**Comment by leseb** [comment]
test this please

**Comment by leseb** [comment]
test this please

**Comment by leseb** [comment]
That's a tricky one and the non-indent from Jinja makes it even worse.
@rootfs, @andrewschoen would you mind having a look since you made some of the changes in here too :)

Thanks!

**Comment by leseb** [comment]
Currently hitting: 
`{"failed": true, "msg": "ERROR! template error while templating string: Encountered unknown tag 'endif'."}`

File: roles/ceph-common/templates/ceph.conf.j2 [hunk 1]
```diff
@@ -9,58 +9,54 @@ auth client required = none
...
 {% endif %}
 {% elif (hostvars[host]['monitor_interface'] is defined and hostvars[host]['monitor_interface'] != "interface") or monitor_interface != "interface" %}
 {% include 'mon_addr_interface.j2' %}
 {% else %}
 {% include 'mon_addr_address.j2' %}
 {% endif %}
-{% endfor %}
+
+fsid = {{ fsid }}
+{% endif %}
```
Comment (andrewschoen) [inline roles/ceph-common/templates/ceph.conf.j2]: Is this `` {% endif %}`` needed?

Comment (leseb) [inline roles/ceph-common/templates/ceph.conf.j2]: you're right! I have other issues now :)

**Comment by leseb** [comment]
test this please

**Comment by leseb** [comment]
Right, this seems to work for bare metal deployments.

@andrewschoen would you mind testing this patch with your 'include'?
@rootfs would you mind testing this patch for containers?

Thanks in advance guys!

**Comment by leseb** [comment]
test this please

**Comment by leseb** [comment]
test this please

**Comment by leseb** [comment]
test this please

**Comment by leseb** [comment]
@rootfs, @andrewschoen up?

**Comment by leseb** [comment]
test this please

**Comment by leseb** [comment]
test this please

**Comment by leseb** [comment]
Guys, i'd like to have this one merged at some point, @rootfs, @andrewschoen 
If I don't get a review from you and I'll merge it this week, sorry if I'm breaking anything for `monitor_address` and containerized deployments... :-/

File: roles/ceph-common/templates/ceph.conf.j2 [hunk 2]
```diff
@@ -8,59 +8,62 @@ auth service required = none
...
-# user address from interface {{ ceph_mon_docker_interface }}
-mon addr = {{ hostvars[host][interface]['ipv4']['address'] }}
- {% elif hostvars[host]['monitor_address'] is defined %}
- # use host monitor address
-mon addr = {{ hostvars[host]['monitor_address'] }}
- {% elif monitor_address != "0.0.0.0" %}
- # use group_var monitor address
-mon addr = monitor_address
-{% endif %}
+mon_host = {% if groups[mon_group_name] is defined %}{% for host in groups[mon_group_name] %}{{ hostvars[host]['ansible_' + monitor_interface]['ipv4']['address'] }}{% if not loop.last %},{% endif %}{% endfor %}{% endif %}
```
Comment (danielin917) [inline roles/ceph-common/templates/ceph.conf.j2]: Why is mon_host defined here when it redefined at the line 28 block

Comment (leseb) [inline roles/ceph-common/templates/ceph.conf.j2]: There are 2 different conditions, for containerized and not containerized deployment.

**Pull request merged**